src/transactions/TransactionUtils.cpp

```diff
@@ -497,15 +498,15 @@ getMaxAmountReceive(LedgerTxnHeader const& header, LedgerEntry const& le)
     }
     if (le.data.type() == TRUSTLINE)
     {
-        int64_t amount = 0;
-        if (isAuthorized(le))
+        auto const& tl = le.data.trustLine();
+        int64_t amount = tl.limit - tl.balance;
+        if (header.current().ledgerVersion >= 10)
```

Comment:
I am not sure I understand the behavior for protocol 10+:
if not authorized, shouldn't we always return 0? Right now it's going to return whatever `tl.limit` is.

Or is it a situation where "we know" that the caller will never call this function if `isAuthorizedToMaintainLiabilities()` is `false`? Which is at best error prone.

If so, I'd rather keep the old code (just use `isAuthorizedToMaintainLiabilities` instead of `isAuthorized`)

Reply:
Yeah this is assuming the authorization was already checked. I'll change this back to the old code and use `isAuthorizedToMaintainLiabilities`.

Reply:
Yeah this is assuming the authorization was already checked. I'll change this back to the old code and use `isAuthorizedToMaintainLiabilities`.